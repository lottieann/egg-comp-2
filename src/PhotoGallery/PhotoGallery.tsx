import * as React from 'react'

import Link from 'next/link'
import Image from 'next/image'

import styled from 'styled-components'

export interface PhotoGalleryProps {
  entryNumber: number;
  title: string;
  thumbnail: string;
  photo1?: string;
  photo2?: string;
  photo3?: string;
  videoLink?: string;
  additional?: string;
}

const Container = styled.div`
  margin-top: 15vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

const TitleWrapper = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
`

const Title = styled.h1`
  font-family: Helvetica;
  font-size: 30px;
  text-align: center;
`

const StyledAnchor = styled.a`
  font-family: Helvetica;
  text-decoration: none;
  cursor: pointer;
  margin-top: 2vh;
`

const PhotoContainer = styled.div`
  dispaly: flex;
  flex-directon: column;
  justify-content: center;
  align-items: center;
`

const PhotoWrapper = styled.div`
  position: relative;
  height: auto;
  width: auto;
  margin: 20px;

  @media screen and (max-width: 600px) {
    width: 90vw;
    margin: 20px;
  }
`

const VideoWrapper = styled.div`
  position: relative;
  width: 90vw;
  margin: 20px;

  @media screen and (max-width: 600px) {
    height: 60vh;
    width: 90vw;
    margin: 20px;
  }
`

export const PhotoGallery = (props: PhotoGalleryProps) => {
  const getPhoto = (photoSrc: string) => {
    return (
      <PhotoWrapper>
        <Image
          src={`/images/${photoSrc}.jpg`}
          alt={`${props.title}`}
          height={600}
          width={800}
          layout="intrinsic"
        />
      </PhotoWrapper>
    )
  }

  return (
    <Container>
      <TitleWrapper>
        <Link href="/gallery">
          <StyledAnchor>Return to Gallery</StyledAnchor>
        </Link>
        <Title>{props.title}</Title>
      </TitleWrapper>
      {props.title === "Jail Eggsotic" && <p>You can't see this video because of a copyright issue with YouTube</p>}
      <PhotoContainer>
        {props.photo1 && getPhoto(props.photo1)}
        {props.photo2 && getPhoto(props.photo2)}
        {props.photo3 && getPhoto(props.photo3)}
        {props.videoLink && (
          <VideoWrapper>
            <iframe
              src={props.videoLink}
              style={{
                position: 'relative',
                height: '60vh',
                width: '90vw',
                margin: '20px',
              }}
            />
          </VideoWrapper>
        )}
        {props.additional && (
          <StyledAnchor href={`${props.additional}`}>Please click here to interact with this entry</StyledAnchor>
        )}
      </PhotoContainer>
    </Container>
  )
}
