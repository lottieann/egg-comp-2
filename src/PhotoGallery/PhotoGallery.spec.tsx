import * as React from 'react'

import 'jest-styled-components'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { PhotoGallery, PhotoGalleryProps } from './PhotoGallery'

const entries = [
  {
    entryName: 'Test Entry',
    entryNumber: 0,
    photo1: 'egg-comp-app/public/0.jpg',
    photo2: 'egg-comp-app/public/1.jpg',
    photo3: 'egg-comp-app/public/3.jpg',
    vote1: '',
    vote2: '',
    vote4: '',
    vote3: '',
    vote5: '',
  },
  {
    entryName: 'Test Entry 1',
    entryNumber: 1,
    photo1: 'egg-comp-app/public/0.jpg',
    photo2: 'egg-comp-app/public/1.jpg',
    photo3: 'egg-comp-app/public/3.jpg',
    vote1: '',
    vote2: '',
    vote4: '',
    vote3: '',
    vote5: '',
  },
  {
    entryName: 'Test Entry 2',
    entryNumber: 2,
    photo1: 'egg-comp-app/public/0.jpg',
    photo2: 'egg-comp-app/public/1.jpg',
    photo3: 'egg-comp-app/public/3.jpg',
    vote1: '',
    vote2: '',
    vote4: '',
    vote3: '',
    vote5: '',
  },
]

const setup = (props: PhotoGalleryProps) => {
  return <PhotoGallery {...props} />
}

describe('PhotoGallery', () => {
  it('should render PhotoGallery correctly', () => {
    const { container } = render(
      setup({
        entries,
        entryIndex: {
          "entry": "1"
        },
      }),
    )
    expect(container).toMatchSnapshot()
  })
})
