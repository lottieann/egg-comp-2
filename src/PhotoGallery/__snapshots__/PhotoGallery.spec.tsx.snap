// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGallery should render PhotoGallery correctly 1`] = `
.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 20vh;
}

.c4 p {
  font-family: Helvetica;
  font-size: 18px;
  text-align: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  width: 0;
  height: 0;
  border-top: 30px solid transparent;
  border-bottom: 30px solid transparent;
  border-left: 30px solid grey;
  border-right: transparent;
  background-color: transparent;
  margin-left: 10px;
}

.c6 {
  height: 30px;
  width: 30px;
  background-color: transparent;
}

.c0 {
  margin-left: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  font-family: Helvetica;
  font-size: 30px;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        class="c2"
      >
        Return to Gallery
      </a>
      <h1
        class="c3"
      >
        Entry 
        1
      </h1>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
          data-testid="previous-button-space"
        />
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <img
            alt="egg-0"
            data-testid="image-0"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
        <button
          class="c7"
          data-testid="next-button"
        />
      </div>
      <p>
        Please enjoy some photos from previous years. Return here Easter Saturday to see 2021 hopefuls!
      </p>
    </div>
  </div>
</div>
`;
